osxkeychain.go

```diff
@@ -108,32 +109,50 @@ func (ke keychainError) Error() string {
 	return fmt.Sprintf("keychainError with unknown error code %d", C.OSStatus(ke))
 }
 
-func AddGenericPassword(attributes *GenericPasswordAttributes) error {
-	if err := attributes.CheckValidity(); err != nil {
-		return err
+func AddGenericPassword(attributes *GenericPasswordAttributes) (err error) {
+	if err = attributes.CheckValidity(); err != nil {
+		return
 	}
 
-	serviceName := C.CString(attributes.ServiceName)
-	defer C.free(unsafe.Pointer(serviceName))
+	var serviceNameString C.CFStringRef
+	if serviceNameString, err = _UTF8StringToCFString(attributes.ServiceName); err != nil {
+		return
+	}
+	defer C.CFRelease(C.CFTypeRef(serviceNameString))
 
-	accountName := C.CString(attributes.AccountName)
-	defer C.free(unsafe.Pointer(accountName))
+	var accountNameString C.CFStringRef
+	if accountNameString, err = _UTF8StringToCFString(attributes.AccountName); err != nil {
+		return
+	}
+	defer C.CFRelease(C.CFTypeRef(accountNameString))
 
-	password := unsafe.Pointer(C.CString(attributes.Password))
-	defer C.free(password)
+	dataBytes := C.CFTypeRef(bytesToCFData(attributes.Password))
```

Comment:
nit: i prefer to have dataBytes be of type CFData and to have CFRelease and the map value cast it